### ISSUE TITLE
Use different NewRelic version or keep it up-to-date

### ISSUE THREAD

**Issue by respinoza** [cloudfoundry/php-buildpack#215]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?

CF version 2.65.0, cli is 6.22.2+a95e24c-2016-10-27

What version of the buildpack you are using?

v4.3.35

If you were attempting to accomplish a task, what was it you were attempting to do?

Use a different version of new relic

What did you expect to happen?

I expected to change the NEWRELIC_VERSION to install a different version

What was the actual behavior?

No effect, looking at the source code it seems it gets the default version from the manifest and there is no way to override it?

I also tried adding a new extensions based on the current newrelic extensio but changing the version to 6.9.0.182 leads to this

`ERR RuntimeError: Could not download dependency: https://download.newrelic.com/php_agent/archive/6.9.0.182/newrelic-php5-6.9.0.182-linux.tar.gz`

Note that the URL is valid.

Anyway, it would be nice to use a different version if possible without the need to build the php package.
We have some applications running with frameworks not supported in the current version (ZF3, Laravel, etc.)

Thank you

Please confirm where necessary:
* [ ] I have included a log output
* [ ] My log includes an error message
* [ ] I have included steps for reproduction

**Comment by sclevine** [comment]
@dmikusa-pivotal how have we kept New Relic up-to-date in the past? Do the New Relic folks PR new versions? Should we set up a dependency notification for this and bump it ourselves?

**Comment by dmikusa-pivotal** [comment]
@sclevine - I think we were pulling in new versions as I was the one that originally added NR support.  I don't know how consistently new versions were pulled in though.  A dependency notification would probably be the way to go.

**Comment by respinoza** [comment]
@dmikusa-pivotal @sclevine Do you need more information from me?

**Comment by respinoza** [comment]
Is there any estimate for this? Thank you

**Comment by sclevine** [comment]
Apologies for the delay @respinoza. Thank you for opening this issue. We have all the information from you that we need.

The New Relic agent will be updated in the next release of the buildpack, which I predict will be available within two weeks. After that, subsequent releases of the buildpack should always contain the most recent version of the agent.

**Comment by respinoza** [comment]
@sclevine great, thank you.

**Comment by mfeltscher** [comment]
There is another issue with the default version (`6.3.0.161`) of the newrelic agent you're currently shipping: It is not compatible with PHP 7.1 which is part of the more recent releases of this buildpack. PHP 7.1 support got added in [version 6.8.0.177](https://docs.newrelic.com/docs/release-notes/agent-release-notes/php-release-notes/php-agent-680177) of the newrelic agent. 

I guess it is safe to upgrade the agent to a version >7 - see https://docs.newrelic.com/docs/release-notes/agent-release-notes/php-release-notes/php-agent-700186

**Comment by mfeltscher** [comment]
Today [v4.3.37](https://github.com/cloudfoundry/php-buildpack/releases/tag/v4.3.37) of this buildpack got released. Unfortunately newrelic remained at `6.3.0.161`. Is there a reason for that?

**Comment by sclevine** [comment]
Our pipeline updated several PHP versions in the buildpack automatically. Instead of sitting on those versions until the New Relic version update story was complete, we opted to release the buildpack without the New Relic update. We will cut a release as soon as that story is complete.

Please follow [this story](https://www.pivotaltracker.com/story/show/147822451).

**Comment by mfeltscher** [comment]
@sclevine I kind of get that but as of right now, you're shipping versions that don't work, since newrelic is incompatible with PHP 7.1 . This breaks existing applications and therefore should be fixed before shipping new broken releases IMO.

**Comment by sclevine** [comment]
Our release process is highly automated. Releasing a new patch version of the buildpack with updated PHP versions (that other users may be waiting on) does not impact our ability to update New Relic and configure a pipeline to keep it up-to-date.

The included version of New Relic never worked with PHP 7.1, so while I would consider this a bug that we should fix soon, I don't consider it a regression. The PHP 7.0 line is still available and up-to-date. Existing apps that use New Relic would need to switch from the PHP 7.0.x version line to the PHP 7.1.x version line to fail staging. Users should not expect to be able to switch version *lines* without experiencing changes in behavior.

**Comment by mfeltscher** [comment]
I totally get that.

Nevertheless, I created a pull request in order to upgrade newrelic: #217

**Comment by sclevine** [comment]
New Relic updated to v7 in the latest release. Future versions of New Relic will be added to the buildpack automatically by the pipeline. Apologies for overlooking this dependency.
